syntax = "proto3";

package tc;

enum RelayMessageType {
  kRelayHello = 0;
  kRelayHeartBeat = 10;
  kRelayTargetMessage = 20;
}

message RelayHello {

}

message RelayHeartBeat {

}

message RelayTargetMessage {
  string room_id = 10;
  string remote_client_id = 20;
  string payload = 30;
}

message RelayMessage {
  string client_id = 10;
  string device_id = 20;
  RelayMessageType type = 30;
  RelayHello hello = 40;
  RelayHeartBeat heartbeat = 50;
  RelayTargetMessage relay = 60;
}